content/blog/pulumi-go-provider-v1/index.md

```diff
@@ -0,0 +1,280 @@
+func (HelloWorld) Create(
+	ctx context.Context, name string, input HelloWorldArgs, preview bool,
+) (string, HelloWorldState, error) {
+	state := HelloWorldState{HelloWorldArgs: input}
+	if preview {
+		return name, state, nil
+	}
+	state.Message = fmt.Sprintf("Hello, %s", input.Name)
+	if input.Loud != nil && *input.Loud {
+		state.Message = strings.ToUpper(state.Message)
+	}
+	return name, state, nil
+}
+```
+
+This minimal code creates a fully-functional resource that your teams can use in their infrastructure code! The SDK handles all the complex gRPC communication with the Pulumi engine and automatically generates the necessary schema for multi-language support.
+
+For detailed step-by-step instructions on building your own provider, check out our [Build a Provider guide](/docs/iac/using-pulumi/extending-pulumi/build-a-provider/).
+
+## Confidence Through Comprehensive Testing
+
+One of the biggest challenges with custom infrastructure providers is ensuring they're reliable. The SDK's built-in testing framework solves this with targeted capabilities:
+
+- **Lifecycle Testing**: Verify your resources' CRUD operations function correctly
+- **Component Testing**: Validate complex resource compositions with mock capabilities
+- **Integration Testing**: Ensure your provider works seamlessly with the Pulumi ecosystem
+
+Here's how easy it is to test your custom resource:
+
+```go
+func TestHelloWorldLifecycle(t *testing.T) {
+	server, err := integration.NewServer(t.Context(), "greetings", semver.Version{Minor: 1},
+		integration.WithProvider(provider()),
+	)
+	require.NoError(t, err)
+
+	// Test the lifecycle methods of the HelloWorld resource.
+	integration.LifeCycleTest{
+		Resource: "pkg:index:HelloWorld",
+		Create: integration.Operation{
+			Inputs: property.NewMap(map[string]property.Value{
+				"name": property.New("tester"),
+				"loud", property.New(true),
+			}),
+			Hook: func(inputs, output property.Map) {
+				message := output.Get("message").AsString()
+				assert.Equal(t, "Hello, tester", message)
+			},
+		},
+	}.Run(t, server)
+}
+```
+
+Need help debugging your provider during development? Our [Debugging Providers guide](/docs/iac/using-pulumi/extending-pulumi/debugging-providers/) walks you through common troubleshooting techniques and tools.
+
+## Standardize Infrastructure with Powerful Component Resources
+
+Beyond simple resources, providers can also define **Component Resources**. Components are higher-level abstractions that encapsulate multiple child resources working together to implement a specific capability. They are perfect for codifying organizational standards and best practices into reusable (cross-language!) building blocks.
+
+The Pulumi Provider SDK v1.0.0 introduces support for defining components using a **functional style**, similar to how you might define a local component within a single program. This provides a streamlined way to build components. Here's an example using the functional style:
```

Comment:
```suggestion
The Pulumi Go Provider SDK introduces support for defining components using a **functional style**, similar to how you might define a local component within a single program. This provides a streamlined way to build components. Here's an example using the functional style:
```

Reply:
In general, I'm confused on whether you want to call this the "Go Provider SDK" or drop the word "Go" ... this draft is about 50/50!   My suggestion is pick one, use it everywhere, and drop the version (since 1.1 will have all the same attributes, etc.)